src/procstat.c

```diff
@@ -313,17 +326,32 @@ static void fuse_write(fuse_req_t req, fuse_ino_t ino, const char *buf,
 	int error;
 	struct procstat_control *control = (struct procstat_control *)ino;
 	struct procstat_item *item = (struct procstat_item *)ino;
+	struct procstat_file *file;
+	int num_objects;
+
+
+	if (item_type_control(item)) {
+		error = control->callback(control->private, buf, size, off);
+		if (!error)
+			fuse_reply_write(req, size);
+		else
+			fuse_reply_err(req, error);
+		return;
+	}
 
-	if (!item_type_control(item)) {
+	file = fuse_inode_to_file(ino);
+	if (!file->writer) {
 		fuse_reply_err(req, EIO);
 		return;
 	}
 
-	error = control->callback(control->private, buf, size, off);
-	if (!error)
+	num_objects = file->writer(file->private, file->arg, (char *)buf, size);
+	/* we currently only support single format parameter */
```

Comment:
can you elaborate about those objects? what are they? num of format character in the format string (i.e. "%f %d" => 2 objects)?
if so, can't we force this when setting the formatter?

Reply:
yes currently only single "variable" is supported